server/src/main/java/org/apache/cloudstack/diagnostics/RetrieveDiagnosticsServiceImpl.java

```diff
@@ -610,7 +651,212 @@ public void run() {
     }
 
     @Override
-    public void cleanupDiagnostics() {
+    public Answer cleanupDiagnostics(DeleteZipCommand cmd) {
```

Comment:
Is this supposed to run in the server or in the agent?
if in the agent it shouldn't be implemented here.
if on the management server there is no need for the Command/Answer pattern
Can you explain your thinking here?